notes/programming-languages/python/packages/webscraping.md

```diff
@@ -0,0 +1,74 @@
+# Python Language Overview
+
+### The `requests` and `BeautifulSoup` Modules
+
+The `requests` and `BeautifulSoup` modules are useful tools for web scraping in python. `requests` connects to and pulls raw data from a web page, while `BeautifulSoup` provides a simple way of traversing a site's html code to pull the parts that you need.
+
+Reference:
+
+  + http://docs.python-requests.org/en/master/
+  + https://www.crummy.com/software/BeautifulSoup/bs4/doc/
+
+> NOTE: `requests` supports Python 2.6–2.7 & 3.3–3.7; `BeautifulSoup` should support Python 2.7 and Python 3.2+
+
+First install the packages using pip, if necessary:
+
+```` sh
+# install requests
```

Comment:
consider omitting commented-out lines in this code block because the preceding markdown is sufficient.